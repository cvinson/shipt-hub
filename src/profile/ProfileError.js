import React from 'react';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';

const ProfileError = ({ errorType }) => {
  let message;

  switch (errorType) {
    case 'notFound':
      message = {
        headline: 'Oh no! We couldn\'t find that user!',
        subtext: 'Please update your search and try again.'
      };
      break;
    case 'rateLimit':
      message = {
        headline: 'Oh no! We\'ve been rate limited!',
        subtext: 'Please try again in a few minutes.'
      };
      break;
    default:
      message = {
        headline: 'An error occurred!',
        subtext: ''
      };
  }

  return (
    <Paper className="errorContainer" zDepth={2}>
      <h3>{message.headline}</h3>
      <p>{message.subtext}</p>
    </Paper>
  );
};

ProfileError.propTypes = {
  errorType: PropTypes.string
};

export default ProfileError;
