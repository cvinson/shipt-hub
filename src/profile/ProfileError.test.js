import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileError from './ProfileError';

describe('<ProfileError />', () => {
  it('renders the default error type correctly', () => {
    const wrapper = shallow(<ProfileError />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Not Found error type correctly', () => {
    const wrapper = shallow(<ProfileError errorType="notFound" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Rate Limit error type correctly', () => {
    const wrapper = shallow(<ProfileError errorType="rateLimit" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
