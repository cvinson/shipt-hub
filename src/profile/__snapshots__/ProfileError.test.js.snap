// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileError /> renders Not Found error type correctly 1`] = `
<Paper
  circle={false}
  className="errorContainer"
  rounded={true}
  transitionEnabled={true}
  zDepth={2}
>
  <h3>
    Oh no! We couldn't find that user!
  </h3>
  <p>
    Please update your search and try again.
  </p>
</Paper>
`;

exports[`<ProfileError /> renders Rate Limit error type correctly 1`] = `
<Paper
  circle={false}
  className="errorContainer"
  rounded={true}
  transitionEnabled={true}
  zDepth={2}
>
  <h3>
    Oh no! We've been rate limited!
  </h3>
  <p>
    Please try again in a few minutes.
  </p>
</Paper>
`;

exports[`<ProfileError /> renders the default error type correctly 1`] = `
<Paper
  circle={false}
  className="errorContainer"
  rounded={true}
  transitionEnabled={true}
  zDepth={2}
>
  <h3>
    An error occurred!
  </h3>
  <p />
</Paper>
`;
